src/interceptors/XMLHttpRequest/utils/createResponse.ts

```diff
@@ -5,9 +5,13 @@ export function createResponse(
   request: XMLHttpRequest,
   responseBody: Uint8Array
 ): Response {
-  return new Response(responseBody, {
-    status: request.status,
-    statusText: request.statusText,
+  // this fix will help to return Error when status is 0
+  const {status} = request;
+  const body = status === 0 ? 'Error' : responseBody;
+  const req = status === 0 ? {status: 500, statusText: 'Internal Server Error'} : request;
```

Comment:
I can also update the condition to use `!status` instead of `status === 0`

Reply:
We should go from the intention here: the intention is to call `createRequest` with a _settled_ request. Settled requests have a non-0 status code. So, in the context of this function, receiving a 0 status code is a breach of the intention, and so this function must throw on those situations. 

You can use the `outvariant` package used in this library to throw on such conditions:

```js
import { invariant } from 'outvariant'

function createResponse(request, responseBody) {
  invariant(request.status !== 0, 'Failed to create a Fetch API Response out of XMLHttpRequest: received a non-settled request (status code 0)')
}
```

This will not solve the issue but it will guard against this function being called improperly.

Reply:
To solve the issue, let's look at _why_ `createResponse` is called in here:

https://github.com/mswjs/interceptors/blob/d73624b222fd80a291486e13eb368d440bf21816/src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts#L454

From the code, you can see that this only happens on a _bypassed request_. If you have a reproduction scenario (where you're taking the error screenshot), link the `@mswjs/interceptors` package to that application and set up some logging/`debugger`s to see _why_ that function is called while `this.status` is still `0` in XMLHttpRequestOverride's context. Perhaps it's a race condition? Maybe we are not awaiting things correctly. 

My suspicion would be that if `createResponse()` is called in the case of a _request error_ (which still fires the `loadend` listener), maybe we're not propagating the error's status onto XMLHttpRequestOverride instance and it remains 0? 🤔

Reply:
@kettanaito - I'll give your suggested solution a shot. I used yalc to integrate this package into my project and came up with the solution described in this PR.

Reply:
Let me know once you come to it. I can help you rebase against the latest main, there've been some changes concerning XHR interception.

Reply:
Sure, I'll work this weekend